engine/hexen2/Makefile

```diff
@@ -992,9 +992,11 @@ USE_CDAUDIO=no
 endif
 
 ifndef DEBUG
-CFLAGS  += -O3
+#CFLAGS  += -O3
```

Comment:
`-O3` is bad with gcc6?

Reply:
Not gcc6 specifically, but I found that O3 bloats up the code in places, so it's probably better to stick with O2 which fits in the cache better.

Reply:
We do `-O3` only for gcc2.95 / aos3, I wonder why we did that back at the time..

Reply:
It was probably an attempt to make it to run faster, but with O3 you start to hit diminishing returns, and in some cases it can do more harm than good.